games/ferium/Portfile

```diff
@@ -0,0 +1,337 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cargo 1.0
+
+name                ferium
+version             4.7.0
+maintainers         {johnlindop.com:git @JLindop} openmaintainer
+revision            0
+
+description         CLI program for downloading and updating Minecraft mods and modpacks.
+
+long_description    Ferium is a CLI tool for downloading and updating Minecraft mods from Modrinth \
+                    CurseForge, and GitHub Releases, and modpacks from Modrinth and CurseForge. \
+
+categories          games
+homepage            https://github.com/gorilla-devs/ferium
+license             MPL-2
+
+checksums           v${version}.tar.gz \
+                    rmd160 2cf1d569e05462e80839b9b9e1b67663ba9e30c3 \
+                    sha256 a7804e44d1949e3fe2c5e875d7d70fc6d2ecd16cbcb65619b9ad4f61ac3b1887 \
+                    size 6899654
+
+master_sites        https://github.com/gorilla-devs/ferium/archive/refs/tags/
```

Comment:
Thanks for this submission @JLindop.

This Portfile will greatly benefit from the `github` portgroup, which lets you use `github.setup` to not have to specify `master_sites`, `distfiles`, etc.  You can see an example of a `cargo` port using the `github` portgroup in various places, one example being `sysutils/hyperfine`.

Reply:
I initially tried using the GitHub portgroup, but I couldn't get it to work. The source code isn't included in the GitHub releases, so I assume that I want `github.tarball_from tarball`, but the URL it generates (`https://codeload.github.com/gorilla-devs/ferium/legacy.tar.gz/v?dummy=`) always returns a 404.

This is how I configured the GitHub portgroup:
```tcl
PortGroup github    1.0

github.setup        gorilla-devs ferium v
github.tarball_from tarball
```

Reply:
You can try `archive` instead of `tarball`

Reply:
Here is documentation on the portgroup: https://guide.macports.org/chunked/reference.portgroup.html

Reply:
I have seen that documentation, but perhaps I didn't understand it correctly.

Using `archive` generates the URL `https://github.com/gorilla-devs/ferium/archive/v`, but even if I manually correct it to `https://github.com/gorilla-devs/ferium/archive/v4.7.0`, it still returns a 404.

GitHub's [recommended URL for downloading tarballs by git tag](https://docs.github.com/en/repositories/working-with-files/using-files/downloading-source-code-archives) is `https://github.com/github/codeql/archive/refs/tags/<tag>`, but there seems to be no way to use that URL in the GitHub portgroup, which is why I set it manually.